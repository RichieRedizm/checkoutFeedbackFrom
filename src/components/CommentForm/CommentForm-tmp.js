import React from 'react';

class CommentForm extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      error: '',

      comment: {
				name: '',
				email: '',
				rating: '',
        message: ''
      }
    };

    this.handleFieldChange = this.handleFieldChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * Handle form input field changes & update the state
   */
  handleFieldChange = event => {
    const { value, name } = event.target;

    this.setState({
      ...this.state,
      comment: {
        ...this.state.comment,
        [name]: value
      }
    });
  };

  /**
   * Form submit handler
   */
  onSubmit(e) {
		e.preventDefault();

    if (!this.isFormValid()) {
      this.setState({ error: "Name and Rating are required fields." });
      return;
    }

		let { comment } = this.state;
		comment.rating = parseInt(comment.rating)
		this.props.addComment(comment);
		this.setState({
			comment: {
				name: '',
				email: '',
				rating: '',
        message: ''
      }
		});
  }

	isFormValid() {
    return this.state.comment.name !== "" && this.state.comment.rating !== "";
	}

  renderError() {
    return this.state.error ? (
      <div className='alert'>{this.state.error}</div>
    ) : null;
  }

  render() {
    return (
      <React.Fragment>
        <form method='post' onSubmit={this.onSubmit}>
          <div className='form-group'>
            <input
              onChange={this.handleFieldChange}
              value={this.state.comment.name}
              className='form-field'
              placeholder='Name'
              name='name'
              type='text'
            />
          </div>

          <div className='form-group'>
            <input
              onChange={this.handleFieldChange}
              value={this.state.comment.email}
              className='form-field'
              placeholder='Email'
              name='email'
              type='email'
            />
          </div>

		  		<div className='form-group'>
            <input
              onChange={this.handleFieldChange}
              value={this.state.comment.rating}
              className='form-field'
              placeholder='Rating 1-5'
              name='rating'
							type='number'
							min='1'
							max='5'
            />
          </div>

          <div className='form-group'>
            <textarea
              onChange={this.handleFieldChange}
              value={this.state.comment.message}
              className='form-field'
              placeholder='Comment'
              name='message'
              rows='5'
            />
          </div>

          <div className='form-group'>
            <button className='btn btn-primary form-field'>
              Comment âž¤
            </button>
          </div>

					{this.renderError()}
        </form>
      </React.Fragment>
    );
  }
}

export default CommentForm
