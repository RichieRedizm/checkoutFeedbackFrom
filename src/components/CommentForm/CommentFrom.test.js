import React from 'react'
import Renderer from 'react-test-renderer'
import CommentForm from './CommentForm'


describe('CommentForm', () => {
	it('Should render the comments form', () => {

		  const component = Renderer.create(<CommentForm />)
		  const tree = component.toJSON()

	  expect(tree).toMatchSnapshot()
	})
  })
