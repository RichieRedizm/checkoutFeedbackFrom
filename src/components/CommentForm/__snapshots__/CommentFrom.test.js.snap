// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm Should render a list of comments with default mock data 1`] = `
<form
  method="post"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-field"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-field"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-field"
      max="5"
      min="1"
      name="rating"
      onChange={[Function]}
      placeholder="Rating 1-5"
      type="number"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <textarea
      className="form-field"
      name="message"
      onChange={[Function]}
      placeholder="Comment"
      rows="5"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-primary form-field"
    >
      Comment âž¤
    </button>
  </div>
</form>
`;
